### PR TITLE
move node maintenance operator to its own api group

### PR DESCRIPTION
**Pull request by ksimon1**
https://bugzilla.redhat.com/show_bug.cgi?id=1766522
/cc @yanirq, @tiraboschi, @beekhof 

Signed-off-by: Karel Simon <ksimon@redhat.com>

### CODE REVIEW

**Comment by beekhof** [comment]
Will this break the UI?

**Comment by ksimon1** [comment]
@beekhof It should not break the UI. @tiraboschi  can you please confirm?

**Comment by beekhof** [comment]
Iirc, the ui is enabled based on whether it finds the crd, and since this changes where the crd lives... or am i missing something?

**Comment by jtomasek** [comment]
Right it will break the UI. The CRD model will need to be updated there: https://github.com/openshift/console/blob/master/frontend/packages/metal3-plugin/src/models.ts#L19

**Comment by ksimon1** [comment]
PR with UI change https://github.com/openshift/console/pull/4967

File: deploy/crds/nodemaintenance_cr.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: kubevirt.io/v1alpha1
+apiVersion: nodemaintenance.kubevirt.io/v1alpha1
```
Comment (tiraboschi) [inline deploy/crds/nodemaintenance_cr.yaml:1]: Please notice that in order to go GA this should be at least v1beta1, maybe it's worth to address both in a single PR.

**Comment by ksimon1** [comment]
@beekhof can you please review this PR?

**Comment by beekhof** [comment]
Can we drop the unrelated whitespace changes to the test file, or move them into a different commit please?
Otherwise it looks good

**Comment by ksimon1** [comment]
> Can we drop the unrelated whitespace changes to the test file, or move them into a different commit please?
> Otherwise it looks good

done

**Comment by beekhof** [comment]
/approve

@n1r1 or @yanirq can you confirm with lgtm if you're happy?

**Comment by n1r1** [comment]
PR is already merged but - does it make sense that the yaml files were changed manually to `v1beta1` but there are still many references to `v1alpha1` in the code? 

For example, the APIs are under `v1alpha1` [package](https://github.com/kubevirt/node-maintenance-operator/tree/master/pkg/apis/kubevirt/v1alpha1)

**Pull request merged**